src/server/PreviewServer.ts

```diff
@@ -109,12 +110,17 @@ export class PreviewServer {
       this.log('establishing server connection...')
 
       const connection = this.app.listen(port, host, () => {
-        const { address, port } = connection.address() as AddressInfo
-        const url = `http://${address}:${port}`
+        const { address, port, family } = connection.address() as AddressInfo
+        // IPv6 host requires surrounding square brackets when serialized to URL
+        // note: IPv6 host can take many forms, e.g. `::` and `::1` are both ok
+        const url = `http://${
+          family === 'IPv6' ? `[${address}]` : address
```

Comment:
Nitpick: a rather complex expression. Consider:

```js
const host = family === 'IPv6' ? `[${address}]` : address
const url = `http://${host}:${port}`
```

Reply:
Good idea! I've implemented your suggestion, except I named the variable "`serializedAddress`," which seems more contextually appropriate, especially given the existing `host` variable in the outer scope, set by the second argument to `listen` (default `"localhost"`).

I modified the existing commit and force pushed the change. Assuming the tests pass, I think this PR is ready to merge.